server/qrops.go

```diff
@@ -0,0 +1,273 @@
+package server
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"image"
+	"net/url"
+	"strconv"
+
+	"github.com/yeqown/go-qrcode/v2"
+	"github.com/yeqown/go-qrcode/writer/standard"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts"
+)
+
+type WriterCloserByteBuffer struct {
+	*bytes.Buffer
+}
+
+func (wc WriterCloserByteBuffer) Close() error {
+	return nil
+}
+
+func NewWriterCloserByteBuffer() *WriterCloserByteBuffer {
+	return &WriterCloserByteBuffer{bytes.NewBuffer([]byte{})}
+}
+
+func ToLogoImageFromBytes(imageBytes []byte, padding int) ([]byte, error) {
+	img, _, err := image.Decode(bytes.NewReader(imageBytes))
+	if err != nil {
+		return nil, fmt.Errorf("decoding image failed: %v", err)
+	}
+	paddedImg := images.AddPadding(img, padding)
+	circle := images.CreateCircle(img)
+	centeredImg := images.PlaceCircleInCenter(paddedImg, circle)
+	resultBytes, err := images.EncodePNG(centeredImg)
+	if err != nil {
+		return nil, fmt.Errorf("encoding PNG failed: %v", err)
+	}
+	return resultBytes, nil
+}
+
+func GetLogoImage(multiaccountsDB *multiaccounts.Database, keyUID string, imageName string) ([]byte, error) {
+	var (
+		imageFolderBasePath = "../_assets/tests/"
+		logoFileStaticPath  = imageFolderBasePath + "status.png"
+		padding             int
+		LogoBytes           []byte
+	)
+
+	identityImageObjectFromDB, err := multiaccountsDB.GetIdentityImage(keyUID, imageName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	staticLogoFileBytes, _ := images.ImageToBytes(logoFileStaticPath)
+
+	if identityImageObjectFromDB == nil {
+		padding = GetPadding(staticLogoFileBytes)
+		LogoBytes, err = ToLogoImageFromBytes(staticLogoFileBytes, padding)
+	} else {
+		padding = GetPadding(identityImageObjectFromDB.Payload)
+		LogoBytes, err = ToLogoImageFromBytes(identityImageObjectFromDB.Payload, padding)
+	}
+
+	return LogoBytes, err
+}
+
+func GetPadding(imgBytes []byte) int {
+	const (
+		defaultPadding = 20
+	)
+	size, _, err := images.GetImageDimensions(imgBytes)
+	if err != nil {
+		return defaultPadding
+	}
+	return size / 5
+}
+
+func getQrURL(params url.Values, logger *zap.Logger) string {
+	qrURL, ok := params["url"]
+
+	if !ok || len(qrURL) == 0 {
+		logger.Error("[qrops-error] no qr url provided")
+	}
+
+	decodedURL, err := base64.StdEncoding.DecodeString(qrURL[0])
+
+	if err != nil {
+		logger.Error("[qrops-error] unable to decoding string from base64", zap.Error(err))
+	}
```

Comment:
This seems like the kind of error that should stop the current flow and warn the user / consuming code about.

Reply:
hmmm, now that I think about it, this could be made better.